storage/storage.go

```diff
@@ -65,6 +65,13 @@ type CertAuthStore interface {
 	AssociateCertHash(r *mdm.Request, hash string) error
 }
 
+type CertAuthRetriever interface {
+	// EnrollmentFromHash retrieves an MDM request from a cert hash.
+	// Implementations should return a nil pointer if no result is found.
+	// The ID member ought to be populated when non-nil.
+	EnrollmentFromHash(ctx context.Context, hash string) (*mdm.Request, error)
```

Comment:
Take an `*x509.Certificate` (or `[]byte`) instead of the hash? This will allow the hashing to be internal to the certauth store.

Also consider returning just a string of the enrollmentID (instead of a whole `*mdm.Request` of which we'll only set one member.

Reply:
On second thought I'm not sure the first item is worthwhile. The storage endpoints are designed to take strings representing the hash. Thus to give a certificate we'd have just have to turn around and ask the certauth service middleware anyway.